config/formats.ts

```diff
@@ -2,7 +2,79 @@
 // The rules that formats use are stored in data/rulesets.js
 import {Utils} from './../lib/utils';
 
-export const Formats: (FormatsData | {section: string, column?: number})[] = [
+import { SideFormats } from './custom-formats';
+
+// interface for the builder.
+interface formatSection {
+	section: string
+	column?: number
+	formats: FormatsData[];
+}
+
+// function for merging the two lists
+function merge(main: (FormatsData | {section: string, column?: number})[],
+			   side: (FormatsData | {section: string, column?: number})[]):
+			   (FormatsData | {section: string, column?: number})[] {
+	// result that is return and makes the actual list for formats.
+	var result: (FormatsData | {section: string, column?: number})[] = [];
+
+	// used as a intermediary to build the final list.
+	var build: formatSection[] = [];
+
+	// used to track location to keep formats under their sections.
+	var loc: number = -1;
+
+	// populates the origonal sections and formats easily
+	// there should be no repeat sections at this point.
+	main.forEach(element => {
+		if (element.section) {
+			build.push({section: element.section, column: element.column, formats: []});
+			loc++;
+		} else if (element.name) {
+			build[loc].formats.push(element);
+		}
+	});
+
+	// merges the second list the hard way. Accounts for repeats.
+	side.forEach(element => {
```

Comment:
```suggestion
	side.forEach(element => {
```
As above, for-of instead.

Reply:
I'll admit. I have no idea what the suggested change is here. What is it?

Reply:
Use a `for-of` loop on the array instead of a forEach()